pennylane/devices/default_qutrit_mixed.py

```diff
@@ -85,10 +88,65 @@ def accepted_sample_measurement(m: qml.measurements.MeasurementProcess) -> bool:
     return isinstance(m, qml.measurements.SampleMeasurement)
 
 
+@qml.transform
+def warn_measurement_error_state(
+    tape: qml.tape.QuantumTape,
+) -> tuple[Sequence[qml.tape.QuantumTape], Callable]:
+    """Warns if a measurement is an analytic State or density_matrix measurement that measurement
+    error is not applied.
+
+    Args:
+        tape (QuantumTape, .QNode, Callable): a quantum circuit.
+
+    Returns:
+        qnode (pennylane.QNode) or quantum function (callable) or tuple[List[.QuantumTape], function]:
+
+        The unaltered input circuit. The output type is explained in :func:`qml.transform <pennylane.transform>`.
+    """
+    if not tape.shots:
+        for m in tape.measurements:
+            if isinstance(m, qml.measurements.StateMP):
+                warnings.warn(f"Measurement {m} is not affected by measurement error.")
+
+    return (tape,), null_postprocessing
+
+
+# PR-comment: separate function so that you can use this to redefine the measurment error without needing new function
+def get_measurement_errors(damping_measurement_gammas, trit_flip_measurement_probs):
```

Comment:
This is a separate function so that you can use this to redefine the measurement error without needing new function

Reply:
suggestion: how about making this a transform as well?

Reply:
I don't think it would work well as a transform, it is part of the class so that the measurement_error can be applied after diagonalizing. If it was a transform I'm not sure how you would deal with Hamiltonians with different non commuting measurements and other similar situations.